src/java/org/apache/cassandra/db/SnapshotDetails.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.db;
+
+import java.time.Instant;
+import java.io.File;
+import java.io.IOException;
+import org.apache.cassandra.io.util.FileUtils;
+import java.util.Map;
+import org.apache.cassandra.config.Duration;
+
+
+
+public class SnapshotDetails {
+    public String tag;
+    public String keyspace;
+    public Instant createdAt;
+    public Instant expiresAt;
+
+    public SnapshotDetails(String tag, String keyspace, File manifestFile) {
+        this.tag = tag;
+        this.keyspace = keyspace;
+        try
+        {
+            Map<String, Object> manifest = FileUtils.readFileToJson(manifestFile);
```

Comment:
Don't do file operations inside constructors. Read the manifest outside before constructing `SnapshotDetails`.

Reply:
plus any implementation of file reading should actually treat exceptions somehow and they should not be ignore like below in order to make it production ready.

@fibersel btw I noticed that you are using this formatting ".parse((String)manifest.get(...)".

There should be a space there like this: ".parse((String) manifest). It applies to more places in this PR.

Reply:
@fibersel if you extract the parsing of manifest file outside as advised by Paulo, I would actually do it in such a way that that SnapshotDetails would accept two more parameters - expiresAt and createdAt - in other words, the parser itself would give me this information comfortably so I do not need to deal with that in this constructor.